### PR TITLE
docs: add an operations cleanup design highlights doc

### PR DESCRIPTION
**Pull request by phlogistonjohn**
### What does this PR achieve? Why do we need it?

This is a design highlights doc to help understand what needs to be done to implement #1401 

### Notes for the reviewer

This has been pushed to obnoxx-collab so feel free to make edits directly to the doc there if you have access. Some sections have been marked [TBD] for items I was less certain of.

### CODE REVIEW

File: docs/design/operation-cleanup.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+* Online, after Heketi server is started
+* Online, a running server will periodically check for operations to clean up
+* On demand, when an admin requests for clean ups using `heketi-cli`
+
+
+Offline mode uses an existing Heketi db and executor configuration without
+relying on a running Heketi server. This action can be seen as a high
+priority cleanup driven by administrators or support or as part of a
+disaster recovery scenario. In this mode, the user will run a command
+such as `heketi-cli offline cleanup-operations` which will perform only
```
Comment (obnoxxx) [inline docs/design/operation-cleanup.md]: Should that be `heketi offline cleanup-operations`?

**Comment by phlogistonjohn** [comment]
I plan on squashing when it's time to merge because I used obnoxxx-collab in case others wanted to push edits directly to the PR.

**Comment by obnoxxx** [comment]
@phlogistonjohn writes:
> I plan on squashing when it's time to merge because I used obnoxxx-collab in case others wanted to push edits directly to the PR.

Oh, I did not realize that you pushed to collab. 
When did you think it's time to merge?

AFAIAC, this can be merged once the above is squashed.

Or do you want to get the TBD resolved first?
I think it might be a good thing to track number of attempts to clean up.

**Comment by phlogistonjohn** [comment]
> @phlogistonjohn writes:
> 
> > I plan on squashing when it's time to merge because I used obnoxxx-collab in case others wanted to push edits directly to the PR.
> 
> Oh, I did not realize that you pushed to collab.

No problem, but I did mention it in the notes for reviewer section.

> When did you think it's time to merge?
> 
> AFAIAC, this can be merged once the above is squashed.
> 
> Or do you want to get the TBD resolved first?

Yes.

> I think it might be a good thing to track number of attempts to clean up.

Yeah, more metrics like that can help if something goes wrong and we need info about if things were making progress or not. Right now I'm thinking of a counter for clean-up attempts and a last-clean-up-attempt timestamp. These would be updated each time the cleanup failed.

**Comment by obnoxxx** [comment]
@phlogistonjohn writes:
> Yeah, more metrics like that can help if something goes wrong and we need info about if things were making progress or not. Right now I'm thinking of a counter for clean-up attempts and a last-clean-up-attempt timestamp. These would be updated each time the cleanup failed.

That sounds useful to me.

**Comment by phlogistonjohn** [comment]
OK, I will update the doc soon to have those metrics captured.

**Comment by phlogistonjohn** [comment]
Doc updated. Currently un-squashed so you can see the diffs. If you're happy feel free to squash yourself or let me know and I'll do it.

**Comment by phlogistonjohn** [comment]
I'm not going to bother making centos-ci succeed for this pr. It's just a text document.

**Comment by phlogistonjohn** [comment]
I'm not going to bother making centos-ci succeed for this pr. It's just a text document.

**Pull request merged**